src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,285 @@
+package main.java.gui;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.util.ArrayList;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.WindowConstants;
+import java.awt.event.*;
+
+/**
+ * GUI pop-up responsible for allowing the user to edit and add different rules
+ * for visualizing the excel table.
+ * 
+ * @author Hugo Barroca
+ */
+public class EditRulePopup {
+	private ArrayList<String> ruleMetrics = new ArrayList<String>();
+	private JFrame frame;
+	private JTextField nameText;
+	private JScrollPane metricsScrollpane;
+	private JComboBox<String> condition;
+
+	private JPanel namePanel;
+	private JPanel metricsPanel;
+	private JPanel metricsListPanel;
+	private JPanel addNewMetricPanel;
+	private JPanel controlPanel;
+
+	private final int FRAME_X = 685;
+	private final int FRAME_Y = 300;
+
+	/**
+	 * Constructs and initializes the GUI pop-up.
+	 */
+	public EditRulePopup() {
+		frame = new JFrame("Personalized Rules");
+		frame.setLayout(new BorderLayout());
+		frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+		frame.add(createMainPanel());
+		frame.setLocationRelativeTo(null);
+		frame.setSize(new Dimension(FRAME_X, FRAME_Y));
+		frame.setVisible(true);
+	}
+
+	/**
+	 * Returns the popup's main SWING frame.

```

Comment:
redundant, no point in saying exactly the same 2 lines apart in the same method - I'd remove this line